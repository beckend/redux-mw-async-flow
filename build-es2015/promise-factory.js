"use strict";
const Bluebird = require("bluebird");
exports.createPromise = () => {
    let resolve;
    let reject;
    const promise = new Bluebird((res, rej) => {
        resolve = res;
        reject = rej;
    });
    return {
        promise,
        resolve: resolve,
        reject: reject
    };
};
//# sourceMappingURL=promise-factory.js.map