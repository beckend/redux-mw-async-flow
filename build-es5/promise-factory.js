"use strict";
var Bluebird = require("bluebird");
exports.createPromise = function () {
    var resolve;
    var reject;
    var promise = new Bluebird(function (res, rej) {
        resolve = res;
        reject = rej;
    });
    return {
        promise: promise,
        resolve: resolve,
        reject: reject,
    };
};
//# sourceMappingURL=promise-factory.js.map