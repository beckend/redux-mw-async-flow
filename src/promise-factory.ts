import * as Bluebird from 'bluebird';

export type TResolveFn<TPayload> = (thenableOrResult?: TPayload | Bluebird.Thenable<TPayload>) => void;
export type TRejectFn = (error?: Error | any) => void;

export const createPromise = <TPayload>() => {
  let resolve: any;
  let reject: any;
  const promise = new Bluebird<TPayload>((res, rej) => {
    resolve = res;
    reject = rej;
  });

  return {
    promise,
    resolve: resolve as TResolveFn<TPayload>,
    reject: reject as TRejectFn,
  };
};
