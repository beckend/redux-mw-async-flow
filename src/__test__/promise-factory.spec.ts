import * as Bluebird from 'bluebird';
import { createPromise } from '../promise-factory';

describe('Promise factory', () => {
  it('createPromise - returns payload as expected', () => {
    const payload = createPromise<any>();
    expect(payload.promise)
      .toBeInstanceOf(Bluebird);
    expect(payload)
      .toMatchSnapshot();
  });
});
