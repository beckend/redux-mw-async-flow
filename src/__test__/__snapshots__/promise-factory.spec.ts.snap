// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Promise factory createPromise - returns payload as expected 1`] = `
Object {
  "promise": Object {
    "fulfillmentValue": undefined,
    "isFulfilled": false,
    "isRejected": false,
    "rejectionReason": undefined,
  },
  "reject": [Function],
  "resolve": [Function],
}
`;
